### ISSUE TITLE
Always getting "WARNING: reconfig partially applied; master may malfunction"?

### ISSUE THREAD

**Issue by dankegel** [buildbot/buildbot#4799]
My master doesn't seem to like reconfiguration, so I've been shutting down and restarting instead.  I'm running v2.2.0-133-g5711c7b4d built from source.  (That's April 26, so it's about five weeks old.  Maybe I should update to v2.3.1, but it's not clear from the commit log it'd help.)

When I send it a SIGHUP, I see: 
```
2019-04-27 21:49:53-0700 [-] reconfig is ongoing for 50 s
2019-04-27 21:50:00-0700 [-] initializing www plugin 'waterfall_view'
2019-04-27 21:50:00-0700 [-] initializing www plugin 'console_view'
2019-04-27 21:50:00-0700 [-] initializing www plugin 'grid_view'
2019-04-27 21:50:00-0700 [Broker,22,10.10.130.255] during reconfig:
        Traceback (most recent call last):
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 654, in _runCallbacks
            current.result = callback(current.result, *args, **kw)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 1475, in gotResult
            _inlineCallbacks(r, g, status)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
            result = result.throwExceptionIntoGenerator(g)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
            return g.throw(self.type, self.value, self.tb)
        --- <exception caught here> ---
          File "/home/buildbot/src/buildbot/master/buildbot/master.py", line 397, in doReconfig
            yield self.reconfigServiceWithBuildbotConfig(new_config)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
            result = result.throwExceptionIntoGenerator(g)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
            return g.throw(self.type, self.value, self.tb)
          File "/home/buildbot/src/buildbot/master/buildbot/util/service.py", line 49, in reconfigServiceWithBuildbotConfig
            yield svc.reconfigServiceWithBuildbotConfig(new_config)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
            result = result.throwExceptionIntoGenerator(g)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
            return g.throw(self.type, self.value, self.tb)
          File "/home/buildbot/src/buildbot/master/buildbot/process/measured_service.py", line 31, in reconfigServiceWithBuildbotConfig
            yield super(MeasuredBuildbotServiceManager, self).reconfigServiceWithBuildbotConfig(new_config)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
            result = result.throwExceptionIntoGenerator(g)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
            return g.throw(self.type, self.value, self.tb)
          File "/home/buildbot/src/buildbot/master/buildbot/util/service.py", line 520, in reconfigServiceWithBuildbotConfig
            yield svc.reconfigServiceWithSibling(config_sibling)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
            result = result.throwExceptionIntoGenerator(g)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
            return g.throw(self.type, self.value, self.tb)
          File "/home/buildbot/src/buildbot/master/buildbot/worker/base.py", line 331, in reconfigServiceWithSibling
            yield self.updateWorker()
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
            result = g.send(result)
          File "/home/buildbot/src/buildbot/master/buildbot/worker/base.py", line 538, in sendBuilderList
            d1 = self.attachBuilder(b)
          File "/home/buildbot/src/buildbot/master/buildbot/worker/base.py", line 543, in attachBuilder
            return builder.attached(self, self.worker_commands)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 1613, in unwindGenerator
            return _cancellableInlineCallbacks(gen)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 1529, in _cancellableInlineCallbacks
            _inlineCallbacks(None, g, status)
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 1421, in _inlineCallbacks
            status.deferred.callback(getattr(e, "value", None))
          File "/home/buildbot/.local/lib/python3.6/site-packages/twisted/internet/defer.py", line 459, in callback
            assert not isinstance(result, Deferred)
        builtins.AssertionError:

2019-04-27 21:50:00-0700 [Broker,22,10.10.130.255] WARNING: reconfig partially applied; master may malfunction
```

**Comment by thendrix** [comment]
I guess I'll mention here too.  I was getting partially applied on "larger" master.cfg as well until I just expanded the timeout:

https://github.com/buildbot/buildbot/issues/4030#issuecomment-484663813

Mine take like 20s because they have slack/discord support and everything in the cfg instead of writing it as a plugin.

**Comment by p12tic** [comment]
This issue has been fixed in #5029. and released in v2.5.0.
